resource_detectors/lib/opentelemetry/resource/detectors/auto_detector.rb

```diff
@@ -12,7 +12,8 @@ module AutoDetector
         extend self
 
         DETECTORS = [
-          OpenTelemetry::Resource::Detectors::GoogleCloudPlatform
+          OpenTelemetry::Resource::Detectors::GoogleCloudPlatform,
+          OpenTelemetry::Resource::Detectors::Deployment
```

Comment:
@open-telemetry/ruby-maintainers I believe we discussed breaking this gem up into separate gems and removing the auto-detector. Is that correct?

Should that be done before we merge this PR?

Reply:
I believe it can be done after this PR is merged unless it creates issues